extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js

```diff
@@ -0,0 +1,378 @@
+							this.updateWikiHtml(wikiId, data);
+							this.notificationsCache[wikiId] = data;
+						}
+					})
+				});
+			},
+
+			updateWikiHtml: function (wikiId, data) {
+				var wikiEl = this.$wallNotifications.find('li[data-wiki-id=' + wikiId + ']'),
+					wikiLi = wikiEl.find('.notifications-for-wiki-list');
+
+				wikiLi.html(data.html);
+
+				if (this.wikiShown[wikiId]) {
+					wikiEl.addClass('show');
+				}
+
+				wikiLi.find('time').timeago();
+
+				// hijack links for other wikis - open them in new window
+				if (wikiId !== this.currentWikiId) {
+					wikiEl.find('a').attr('target', '_blank');
+
+					// temporary fix, should be changed on server side
+					wikiEl.find('.read').remove();
+				}
+
+				this.setNotificationsHeight();
+			},
+
+			getLastSeenCount: function () {
+				return $.cookies.get('WallUnreadCount') || 0;
+			},
+
+			setLastSeenCount: function (val) {
+				$.cookies.set('WallUnreadCount', val);
+			},
+
+			proxy: function (func) {
+				return $.proxy(func, this);
+			},
+
+			getUrlParams: function () {
+				var data = {},
+					qs = Wikia.Querystring(),
+					lang, skin;
+
+				skin = qs.getVal('useskin');
+				if (skin) {
+					data.useskin = skin;
+				}
+
+				lang = qs.getVal('uselang');
+				if (lang) {
+					data.uselang = lang;
+				}
+
+				return data;
+			},
+
+			setNotificationsHeight: function () {
+				var isDropdownOpen = this.$wallNotifications.hasClass('show'),
+					height = 0,
+					msgHeight = 0;
+
+				if (isDropdownOpen) {
+
+					if (this.notificationsMarkAsReadHeight === 0) {
+						this.notificationsMarkAsReadHeight = $('.notifications-markasread').outerHeight();
+					}
+
+					height = this.$window.height() - this.globalNavigationHeight - this.notificationsBottomPadding - this.notificationsMarkAsReadHeight;
+					msgHeight = this.$notificationsMessages.height();
+
+					if (!msgHeight) {
+						this.$notificationsContainer = $('#notificationsContainer');
+						this.$notificationsMessages = $('> ul', this.$notificationsContainer);
+						msgHeight = this.$notificationsMessages.height();
+					}
+
+					if (this.notificationsHeaderHeight <= 0) {
+						this.notificationsHeaderHeight = $('.notifications-header', this.$wallNotifications).outerHeight();
+					}
+
+					if (height < msgHeight + this.notificationsHeaderHeight) {
+						this.$notificationsContainer
+							.css('height', height - this.notificationsHeaderHeight)
+							.addClass('scrollable');
+					} else {
+						this.$notificationsContainer.css('height', 'auto').removeClass('scrollable');
+					}
+				}
+			},
+
+			onNotificationsOpen: function () {
+				WallNotifications.$wallNotifications.addClass('show');
+				WallNotifications.setNotificationsHeight();
+				$('#globalNavigation').trigger('notifications-menu-opened');
+			}
+		};
+
+		$(function () {
+			WallNotifications.init();
+
+			// TODO make it pretty
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)